pkg/cmd/pr/shared/templates.go

```diff
@@ -201,14 +282,15 @@ func (m *templateManager) fetch() error {
 	}
 
 	if hasAPI {
-		issueTemplates, err := listIssueTemplates(m.httpClient, m.repo)
+		lister := listIssueTemplates
+		if m.isPR {
+			lister = listPullRequestTemplates
+		}
+		templates, err := lister(m.httpClient, m.repo)
 		if err != nil {
 			return err
 		}
-		m.templates = make([]Template, len(issueTemplates))
-		for i := range issueTemplates {
-			m.templates[i] = &issueTemplates[i]
-		}
```

Comment:
I moved this work into the `lister` functions so that they would both have the same function signatures.